main.js

```diff
@@ -0,0 +1,284 @@
+const Handlebars = require('handlebars');
+const fse = require('fs-extra');
+const path = require('path');
+const yaml = require('js-yaml');
+const docmatter = require('docmatter');
+const Markdown = new require('markdown-it')({
+    html: true, // enable HTML in Markdown
+    typographer: true // have some nice pretty quotes
+}).use(require('markdown-it-highlightjs'), {auto: true, code: false})
+
+const outDir = 'html'; // TODO cannonicalize
+
+fse.mkdirpSync(outDir);
+fse.emptyDirSync(outDir);
+
+compileTemplate = template => Handlebars.compile(template.toString('UTF-8'));
+
+/**
+ * 
+ * @param {string} dirname The directory to read HTML content from
+ * @param {Handlebars.TemplateDelegate} wrapperTemplate Handlebars template to apply to the content
+ * @param {Object} globalContext Context to use for every page
+ * @returns {Promise} no resolve value, wraps around a Promise.all of all of the files
+ */
+function regularDir(dirname, wrapperTemplate, globalContext) {
+    return fse.readdir(dirname).then(listing => {
```

Comment:
As with below, the dir should not be assumed to exist. Error handling is needed for all `readdir`s.